sormas-backend/src/main/java/de/symeda/sormas/backend/sormastosormas/share/outgoing/ShareRequestInfoService.java

```diff
@@ -72,9 +74,16 @@ public List<String> getAllNonReferencedShareRequestInfo() {
 		Join<ShareRequestInfo, SormasToSormasShareInfo> shareRequestJoin = from.join(ShareRequestInfo.SHARES, JoinType.LEFT);
 		cq.select(from.get(ShareRequestInfo.UUID));
 		cq.groupBy(from.get(ShareRequestInfo.ID));
-		cq.having(cb.equal(cb.count(shareRequestJoin), 0));
 
-		return em.createQuery(cq).getResultList();
+		Predicate revokedStatus = cb.equal(from.get(ShareRequestInfo.REQUEST_STATUS), ShareRequestStatus.REVOKED);
```

Comment:
In order to delete the ShareRequestInfos, the getAllNonReferencedShareRequestInfo() will be called before to find the ShareRequestInfos which should be deleted.
![image](https://user-images.githubusercontent.com/47103965/198215494-fe589649-cfe4-44e3-a974-edd63b403779.png)


The ShareRequestInfos till now only retrieved the ShareRequestInfos without joins which meant that they are no longer referenced. In this task the ShareRequestInfos will be still referenced but their status was changed to REVOKED.

Due to above mentioned problem testSormasToSormasShareInfoPermanentDeletion() unit test from CoreEntityDeletionServiceTest failed because the getAllNonReferencedShareRequestInfo did not return the REVOKED ShareRequestInfo as non referenced => it was not deleted

assertEquals(0, getShareRequestInfoService().count());  ( Before the returned count was 1 instead of 0, since the ShareRequestInfo  was not find and deleted).

I'll add a comment in the code too

Reply:
ok thank you!